module_utils/certificate/providers/certmonger.py

```diff
@@ -0,0 +1,142 @@
+import os
+
+from io import StringIO
+
+# pylint: disable=no-name-in-module
+from ansible.module_utils.six.moves.configparser import ConfigParser
+
+from .base import CertificateRequestBaseProvider
+
+
+class CertificateRequestCertmongerProvider(CertificateRequestBaseProvider):
+    """Certmonger provider for certificate Linux System Role."""
+
+    def __init__(self, *args, **kwargs):
+        super(CertificateRequestCertmongerProvider, self).__init__(*args, **kwargs)
+
+        self.requests_dir = "/var/lib/certmonger/requests"  # TODO: Allow to override?
+        self._request_file_path = self._find_certmonger_request_file()
+        self._certmonger_metadata = self._read_certmonger_request_file()
+
+    def _find_certmonger_request_file(self):
+        """Search for certificate metadata in certmonger data directory.
+
+        Looks for the certificate path, file by file inside certmonger data
+        directory.
+
+        Only open files that are 50k or smaller to avoid getting stuck for
+        a long time.
+
+        If the file is found it's immediately. If not found returns None.
```

Comment:
"it's returned immediately"

Reply:
4e74b07b67c935b8d577375e7779606444905190